### PR TITLE
Api game controller

### PR DESCRIPTION
**Pull request by benjaminpayoux**
[Trello](https://trello.com/c/UvcT7HyF/1-05-as-jean-peter-i-would-like-to-see-how-the-battleship-game-could-fit-on-a-mobile-device)

# TODO
- [X] add service to create a new game from the react app
- [X] get a game with its hash

### CODE REVIEW

File: app/config/packages/nelmio_cors.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+nelmio_cors:
```
Comment (Luwangel) [inline app/config/packages/nelmio_cors.yaml:1]: You have to use a plugin to manage cors?

Comment (benjaminpayoux) [reply app/config/packages/nelmio_cors.yaml:1]: Easier than doing it by hand. The header must be returned first on the OPTION request. The lib do it automatically for me.

File: app/src/Controller/Api/GameController.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+use App\Repository\ShipRepository;
+use App\Twig\AppExtension;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Routing\Annotation\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+
+class GameController extends AbstractController
+{
+    /**
+     * Create a game and returns its hash to the player
```
Comment (Luwangel) [inline app/src/Controller/Api/GameController.php]:
```suggestion
     * Create a game and return its hash to the player
```

or __Creates a game and returns__

File: app/src/Controller/Api/GameController.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace App\Controller\Api;
+
+use App\Entity\Game;
+use App\GameManipulator;
+use App\Repository\GameRepository;
+use App\Repository\ShipRepository;
+use App\Twig\AppExtension;
```
Comment (Luwangel) [inline app/src/Controller/Api/GameController.php:9]: Do you need Twig?

File: app/src/Entity/Player.php [hunk 4]
```diff
@@ -65,7 +65,7 @@ public function getShoots(): Collection
         return $this->shoots;
     }
 
-    public function getHash(): string 
+    public function getHash(): ?string
```
Comment (Luwangel) [inline app/src/Entity/Player.php:68]: Sometimes the method returns `null`?

**Review by Luwangel — CHANGES_REQUESTED** [review]

File: app/src/Controller/Api/GameController.php (hunk 3 above)
Comment (benjaminpayoux) [reply app/src/Controller/Api/GameController.php:9]: Yes, for the ships grid attributes. It helps me place them on the board in the frontend.

File: app/src/Entity/Player.php (hunk 4 above)
Comment (benjaminpayoux) [reply app/src/Entity/Player.php:68]: Yes, when I create a game, I init a second player which does not have a hash yet.

**Review by Luwangel — APPROVED** [review]

**Pull request merged**